exporter/stackdriverexporter/stackdriver.go

```diff
@@ -19,96 +19,106 @@ package stackdriverexporter
 	}
-
-	return mExp, doneFn, nil
+	mExp := &stackdriverExporter{exporter: sde}
+
+	return exporterhelper.NewMetricsExporter(
+		cfg,
+		mExp.pushMetricsData,
+		exporterhelper.WithTracing(true),
+		exporterhelper.WithMetrics(true),
+		exporterhelper.WithShutdown(mExp.Shutdown))
 }
 
-func newStackdriverExporter(ProjectID, MetricPrefix string) (*stackdriver.Exporter, error) {
+func getStackdriverOptions(ProjectID, MetricPrefix string) stackdriver.Options {
 	// TODO:  For each ProjectID, create a different exporter
 	// or at least a unique Stackdriver client per ProjectID.
 
-	return stackdriver.NewExporter(stackdriver.Options{
+	return stackdriver.Options{
 		// If the project ID is an empty string, it will be set by default based on
 		// the project this is running on in GCP.
 		ProjectID: ProjectID,
 
 		MetricPrefix: MetricPrefix,
 
-		// Stackdriver Metrics mandates a minimum of 60 seconds for
-		// reporting metrics. We have to enforce this as per the advisory
-		// at https://cloud.google.com/monitoring/custom-metrics/creating-metrics#writing-ts
-		// which says:
-		//
-		// "If you want to write more than one point to the same time series, then use a separate call
-		//  to the timeSeries.create method for each point. Don't make the calls faster than one time per
-		//  minute. If you are adding data points to different time series, then there is no rate limitation."
-		BundleDelayThreshold: 61 * time.Second,
-	})
+		// Set DefaultMonitoringLabels to an empty map to avoid getting the "opencensus_task" label
+		DefaultMonitoringLabels: &stackdriver.Labels{},
+	}
 }
 
-func (sde *stackdriverExporter) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
-	ctx, span := trace.StartSpan(ctx,
-		"opentelemetry.service.exporter.stackdriver.ExportMetricsData",
-		trace.WithSampler(trace.NeverSample()))
-	defer span.End()
+// pushMetricsData is a wrapper method on StackdriverExporter.PushMetricsProto
+func (se *stackdriverExporter) pushMetricsData(ctx context.Context, md consumerdata.MetricsData) (int, error) {
+	return se.exporter.PushMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
+}
 
-	err := sde.exporter.ExportMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
-	if err != nil {
-		span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: err.Error()})
+// TODO(songya): add an interface PushSpanProto to Stackdriver exporter and remove this method
+// pushTraceData is a wrapper method on StackdriverExporter.PushSpans
+func (se *stackdriverExporter) pushTraceData(ctx context.Context, td consumerdata.TraceData) (int, error) {
+	var errs []error
+	var goodSpans []*tracepb.Span
+	for _, span := range td.Spans {
+		sd, err := spandatatranslator.ProtoSpanToOCSpanData(span)
+		if err == nil {
+			se.exporter.ExportSpan(sd)
+			goodSpans = append(goodSpans, span)
+		} else {
+			errs = append(errs, err)
+		}
 	}
 
-	return err
+	return len(td.Spans) - len(goodSpans), oterr.CombineErrors(errs)
```

Comment:
Get rid of GoodSpans and just count the length

Reply:
SG, updated.